clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1281,17 +1259,11 @@ private void recordTopicFetchMetrics(String topic, int bytes, int records) {
                 Map<String, String> metricTags = Collections.singletonMap("topic", topic.replace('.', '_'));
 
                 bytesFetched = this.metrics.sensor(name);
-                bytesFetched.add(this.metrics.metricName("fetch-size-avg",
-                        this.metricGrpName,
-                        "The average number of bytes fetched per request for topic " + topic,
+                bytesFetched.add(this.metrics.metricInstance(metricsRegistry.topicFetchSizeAvg,
```

Comment:
Regarding `MetricNameTemplate`: makes sense.

Regarding `*MetricsRegsitry`, I think even with these classes we cannot centralize them right? think: if we add a new metric in the future, and the developer need to add it both in the `*MetricsRegistry` and here; and if she forgot to add it in this class, nothing will break at compile time, so nothing is programmatically enforced.